packages/flutter_tools/lib/src/artifacts.dart

```diff
@@ -1538,6 +1538,11 @@ class CachedLocalWebSdkArtifacts implements Artifacts {
   }
 
   String _getDartSdkPath() {
+    // If the parent is a local engine, then use the locally built Dart SDK.
+    if (_parent.usesLocalArtifacts) {
+      return _parent.getArtifactPath(Artifact.engineDartSdkPath);
```

Comment:
Just to clarify, this only happens if you're passing both `--local-web-sdk` and `--local-engine`, correct? I just want to make sure this doesn't happen if only `--local-web-sdk` is passed. (Those of us who work on the web engine very rarely actually do a full local engine build, so we don't want to accidentally pick up a stale version of the dart sdk).

Reply:
Yes - if you specify `--local-web-sdk` and `--local-engine`, then this will use the Dart SDK built as part of the local engine.

If you only specify `--local-web-sdk`, then this will maintain the current behavior which uses the prebuilt Dart SDK.

Reply:
Great, thank you!